fuel-relayer/src/pending_events.rs

```diff
@@ -0,0 +1,375 @@
+use std::collections::{HashMap, VecDeque};
+
+use fuel_core_interfaces::relayer::RelayerDb;
+use fuel_tx::{Address, AssetId, Bytes32};
+use fuel_types::Word;
+use tracing::info;
+
+use crate::log::EthEventLog;
+
+pub struct PendingEvents {
+    /// Pendning stakes/assets/withdrawals. Before they are finalized
+    /// it contains every fuel block and its span
+    pending: VecDeque<PendingDiff>,
+    /// This is litlle bit hacky but because we relate validator staking with fuel commit block and not on eth block
+    /// we need to be sure that we are taking proper order of those transactions
+    /// Revert are reported as list of reverted logs in order of Block2Log1,Block2Log2,Block1Log1,Block2Log2.
+    /// I checked this with infura endpoint.
+    bundled_removed_eth_events: Vec<(u64, Vec<EthEventLog>)>,
```

Comment:
Why not use a btree map instead?

Reply:
hm, my first thought, why to use it if it is not needed. But maybe there is something here, i will need to see if there is benefits there.

Reply:
I will probably use that inside the next iteration of Relayer, as I have already started in the other branch.